demo/c/README.md

```diff
@@ -1,33 +1,176 @@
+## Requirements
+
+**For Windows, MingW is required to run the demo.**
+
+The microphone based demo requires [miniaudio](https://github.com/mackron/miniaudio) for accessing microphone audio data.
+
+## Build
+
+### Linux, macOS, Raspberry Pi
+
 ```console
-gcc -std=c99 -O3 -o demo/c/picovoice_demo_mic \
--I sdk/c/include demo/c/picovoice_demo_mic.c -ldl -lasound
+gcc -std=c99 -O3 -o demo/c/picovoice_demo_mic -I sdk/c/include demo/c/picovoice_demo_mic.c -ldl -lpthread -lm
 ```
 
+### Windows
+
+```console
+gcc -std=c99 -O3 -o demo/c/picovoice_demo_mic -I sdk/c/include demo/c/picovoice_demo_mic.c
+```
+
+## Run
+
+Running the executable without any commandline arguments prints the usage info to the console.
+
+For Linux, macOS, and Raspberry Pi:
+
+```console
+./demo/c/picovoice_demo_mic
+usage : ./demo/c/picovoice_demo_mic library_path porcupine_model_path keyword_path porcupine_sensitivity \
+                                    rhino_model_path context_path rhino_sensitivity input_audio_device
+        ./demo/c/picovoice_demo_mic --show_audio_devices
+```
+
+on Windows:
+
+```console
+./demo/c/picovoice_demo_mic.exe
+usage : ./demo/c/picovoice_demo_mic.exe library_path porcupine_model_path keyword_path porcupine_sensitivity \
+                                        rhino_model_path context_path rhino_sensitivity input_audio_device
+        ./demo/c/picovoice_demo_mic.exe --show_audio_devices
+```
+
+To show the available audio input devices, on Linux, macOS, Raspberry Pi run:
+
+```console
+./demo/c/picovoice_demo_mic --show_audio_devices
+```
+
+on Windows run:
+
+```console
+./demo/c/picovoice_demo_mic.exe --show_audio_devices
+```
+
+The following commands start up a microphone audio steam and will wait for the `picovoice` wake word phrase.
+
+### Linux
+
 ```console
 ./demo/c/picovoice_demo_mic \
-${LIBRARY_PATH} \
+sdk/c/lib/linux/x86_64/libpicovoice.so \
 resources/porcupine/lib/common/porcupine_params.pv \
-resources/porcupine/resources/keyword_files/${PLATFORM}/picovoice_${PLATFORM}.ppn \
+resources/porcupine/resources/keyword_files/linux/picovoice_linux.ppn \
 0.5 \
 resources/rhino/lib/common/rhino_params.pv \
-resources/rhino/resources/contexts/${PLATFORM}/coffee_maker_${PLATFORM}.rhn \
+resources/rhino/resources/contexts/linux/smart_lighting_linux.rhn \
 0.5 \
-${INPUT_AUDIO_DEVICE}
+{AUDIO_DEVICE_INDEX}
 ```
 
+### macOS
+
 ```console
-gcc -std=c99 -O3 -o demo/c/picovoice_demo_file \
--I sdk/c/include demo/c/picovoice_demo_file.c -ldl
+./demo/c/picovoice_demo_mic \
+sdk/c/lib/mac/x86_64/libpicovoice.dylib \
+resources/porcupine/lib/common/porcupine_params.pv \
+resources/porcupine/resources/keyword_files/mac/picovoice_mac.ppn \
+0.5 \
+resources/rhino/lib/common/rhino_params.pv \
+resources/rhino/resources/contexts/mac/smart_lighting_mac.rhn \
+0.5 \
+{AUDIO_DEVICE_INDEX}
 ```
 
+### RaspberryPi
+
+Replace `${PROCESSOR}` with one of Raspberry Pi's processor defined [here](../../sdk/c/lib/raspberry-pi) (for Raspberry Pi 4 this would
```

Comment:
one of Raspberry Pi's processor defined [here] => one of the Raspberry Pi processors defined [here]

Reply:
(e.g., for Raspberry Pi 4 this would be "cortex-a72")